### PR TITLE
Bump storeApiVersion from 0.2.0 to 0.3.0

### PR DESCRIPTION
**Pull request by emmberk**
I also removed `/mis` in the paths.

I'd like to get this merged before https://github.com/connexta/ion-store/pull/216 so that PR will be the breaking commit for Transform Team.

### CODE REVIEW

File: src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java [hunk 1]
```diff
@@ -84,32 +84,30 @@ public void store(
    */
   @Override
   @NotNull
-  public RetrieveResponse retrieve(@NotBlank final String key) throws RetrieveException {
-    log.info("Retrieving product in bucket \"{}\" with key \"{}\"", bucket, key);
+  public RetrieveResponse retrieve(@NotBlank final String key) throws RetrieveFileException {
+    log.info("Retrieving item in bucket \"{}\" with key \"{}\"", bucket, key);
```
Comment (jhunzik) [inline src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java]: ✏️ `file`?

Comment (emmberk) [reply src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java]: According to Architecture Team's definitions, `metadata` is not a `file`, and this class is now used for both `file`s and IRM.

Comment (emmberk) [reply src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java]: I just realized that I should update the name of the exception.

Comment (emmberk) [reply src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java]: Sorry, I got confused with another PR that I'm working on. This is only used for `file`s right now. I'll update it to say `file`.

File: README.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ For more information about spotless checks see
     Example configs/store_config.yml:
     ```yaml
     endpointUrl:
-      retrieve: http://localhost:9041/mis/product/
+      retrieve: http://localhost:9041/dataset/
```
Comment (peterhuffer) [inline README.md]: Now that I am reading this in this PR, it seems a little odd we are calling this dataset, but only sending a single file.

Comment (emmberk) [reply README.md]: Actually, it wouldn't "break" our contract with Transform Service.

https://github.com/connexta/ion-transformation-api/blob/7f51d2ab85fb4684e3c71d47d8931d3f3d1242bb/rest/specs/src/main/filtered-resources/openapi/openapi.yaml#L198-L200

So, if the `file` location was something like  `/datatset/<datasetId>/file`, the cst location would have to be `/datatset/<datasetId>/file/metadata/cst`.

File: src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java (hunk 1 above)
Comment (jhunzik) [reply src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java]: The only downside to using `file` to refer to the resource is that technically metadata is a file too.

**Pull request merged**